bgpd/bgp_zebra.c

```diff
@@ -3901,3 +3901,39 @@ static int bgp_opaque_msg_handler(ZAPI_CALLBACK_ARGS)
 stream_failure:
 	return ret;
 }
+
+void bgp_zebra_send_nexthop_label(int cmd, mpls_label_t label,
+				  ifindex_t ifindex, vrf_id_t vrf_id,
+				  enum lsp_types_t ltype, struct prefix *p)
+{
+	struct zapi_labels zl = {};
+	struct zapi_nexthop *znh;
+
+	if (cmd != ZEBRA_MPLS_LABELS_ADD && cmd != ZEBRA_MPLS_LABELS_DELETE)
+		return;
+
+	zl.type = ltype;
+	zl.local_label = label;
+	zl.nexthop_num = 1;
+	znh = &zl.nexthops[0];
+	if (p->family == AF_INET)
+		znh->gate.ipv4 = p->u.prefix4;
+	else
+		memcpy(&znh->gate.ipv6, &p->u.prefix6, sizeof(struct in6_addr));
```

Comment:
IPV6_ADDR_COPY, also, the same for IPV4.

Reply:
ok

Reply:
not done

Reply:
I did it in the original commit. thanks.